### ISSUE TITLE
Syntax Parser doesn't recognize the quote-in-quote syntax

### ISSUE THREAD

**Issue by MacHu-GWU** [bridgecrewio/checkov#446]
**Describe the bug**

For ``checkov -f path-to-terraform-script.tf``, when there's terraform code like this:

```
... {
   SOME_HASH = "${filemd5("${path.module}/templates/some-file.json")}"
}
```

checkov raises:

```
Traceback (most recent call last):
  File "/Users/sanhehu/venvs/python/3.6.2/idm_devops_venv/bin/checkov", line 5, in <module>
    run()
  File "/Users/sanhehu/venvs/python/3.6.2/idm_devops_venv/lib/python3.6/site-packages/checkov/main.py", line 87, in run
    scan_reports = runner_registry.run(external_checks_dir=args.external_checks_dir, files=args.file, guidelines=guidelines)
  File "/Users/sanhehu/venvs/python/3.6.2/idm_devops_venv/lib/python3.6/site-packages/checkov/common/runners/runner_registry.py", line 26, in run
    runner_filter=self.runner_filter)
  File "/Users/sanhehu/venvs/python/3.6.2/idm_devops_venv/lib/python3.6/site-packages/checkov/terraform/runner.py", line 58, in run
    self.check_tf_definition(report, root_folder, runner_filter)
  File "/Users/sanhehu/venvs/python/3.6.2/idm_devops_venv/lib/python3.6/site-packages/checkov/terraform/runner.py", line 80, in check_tf_definition
    definitions_context = parser_registry.enrich_definitions_context(definition)
  File "/Users/sanhehu/venvs/python/3.6.2/idm_devops_venv/lib/python3.6/site-packages/checkov/terraform/context_parsers/registry.py", line 21, in enrich_definitions_context
    definition_blocks_types = {x: definition_blocks_types[x] for x in definition_blocks_types.keys()}
AttributeError: 'NoneType' object has no attribute 'keys'
```

however magically when I change it to this, it works:

```
"${filemd5(${path.module}/templates/some-file.json)}"
```

**Seems like it is the checkov syntax parser is be able to handle double-quote-in-quote syntax. However, based on the official document from terraform website https://www.terraform.io/docs/providers/aws/r/s3_bucket_object.html. The previous one, which is the one I use, is the valid version.**

**To Reproduce**

see above

**Expected behavior**

it should be able to parse the syntax and run the scan.

**Screenshots**
NO

**Desktop (please complete the following information):**
 - OS: Mac
 - Checkov Version 1.0.412

**Comment by metahertz** [comment]
Hi! Thanks for raising this!

Just to clarify (wondering if the docs page you linked has been updated since), it looks like theres no double quoting on the example?

```
resource "aws_s3_bucket_object" "object" {
  bucket = "your_bucket_name"
  key    = "new_object_key"
  source = "path/to/file"

  # The filemd5() function is available in Terraform 0.11.12 and later
  # For Terraform 0.11.11 and earlier, use the md5() function and the file() function:
  # etag = "${md5(file("path/to/file"))}"
  etag = filemd5("path/to/file")
}
```

**Comment by robeden** [comment]
The new parser handles this gracefully.
